### ISSUE TITLE
Can we use SIP via SSL/TLS between drachtio server and legacy PBX?

### ISSUE THREAD

**Issue by calvet27** [davehorton/drachtio-server#109]
Dear Mr. Horton,

We have to get the signaling connection via TLS between our SIP gateway which includes drachtio server and legacy PBX. Is there any advices?
As I know, sofia supports this feature. Please let us get this.

**Comment by davehorton** [comment]
Yes.  In your drachtio.conf.xml you will need to add a contact to listen for tls, something like:
```xml
 <contact>sips:<your-ip>:5061;transport=tls</contact>
```
and you will also need to specify the location of your tls certs:
```
        <tls>
                <key-file>/etc/letsencrypt/live/yourdomain/privkey.pem</key-file>
                <cert-file>/etc/letsencrypt/live/yourdomain/cert.pem</cert-file>
                <chain-file>/etc/letsencrypt/live/yourdomain/chain.pem</chain-file>
        </tls>
```
See [here for more detail](https://github.com/davehorton/drachtio-server/blob/master/drachtio.conf.xml)

Try this and let me know of any issues

**Comment by calvet27** [comment]
> Yes. In your drachtio.conf.xml you will need to add a contact to listen for tls, something like:
> 
> ```
>  <contact>sips:<your-ip>:5061;transport=tls</contact>
> ```
> 
> and you will also need to specify the location of your tls certs:
> 
> ```
>         <tls>
>                 <key-file>/etc/letsencrypt/live/yourdomain/privkey.pem</key-file>
>                 <cert-file>/etc/letsencrypt/live/yourdomain/cert.pem</cert-file>
>                 <chain-file>/etc/letsencrypt/live/yourdomain/chain.pem</chain-file>
>         </tls>
> ```
> 
> See [here for more detail](https://github.com/davehorton/drachtio-server/blob/master/drachtio.conf.xml)
> 
> Try this and let me know of any issues

Thank you sir. ^^ I will try that configuration and then let you know the results.

**Comment by calvet27** [comment]
I'm so sorry to late reply.
Here is the results for above configuration.
Please help me out.

[drachtio_logs_tls_error_20200310.txt](https://github.com/davehorton/drachtio-server/files/4310741/drachtio_logs_tls_error_20200310.txt)

**Comment by davehorton** [comment]
this line
```
adding additional sip address sip:150.150.140.78:5061;transport=tls
nta.c:2273 nta_agent_add_tport() nta: bind(150.150.140.78:5061;transport=tls): Protocol not supported
```
implies you did not create the contact correctly.

In your config, the scheme should be "sips" not "sip".

Also, it seems you are running an older release.  You are running from the develop branch, correct?

**Comment by calvet27** [comment]
Thank you for your reply.

1. From the example config in github, There is only <sip> not <sips>. You mean that I should change it to <sips>. Am I right?
2. As I told you from another issue, I use docker image for drachtio server. Where can I get the latest version of drachtio server as a docker image?

**Comment by calvet27** [comment]
Sorry, I realized what I missed.
I  changed the schema to sips from sip for contact.
And I also changed the docker image as the latest.
Thank you.

**Comment by moonlitknight** [comment]
Hi Dave, first time caller :-) 
Loving drachtio so far

I had the same problem as the OP. Is this a bug on https://drachtio.org/docs/drachtio-server which says ...
```
<!-- listen for secure websockets on specific address and port -->
<contacts>
  <contact>sip:192.168.100.23:443;transport=wss</contact>
</contacts>
```

**Comment by davehorton** [comment]
what happens when you start the drachtio server?  Can you show the drachtio log?

**Comment by davehorton** [comment]
also, when using transport=wss try using `sips:` instead of `sip:`.

**Comment by moonlitknight** [comment]
Hi Dave, as with calvet27, my problem was solved by replacing `sip` with `sips`.
My comment was that the documentation says `sip` so is this a documentation error?

I'm just starting with drachtio and would be happy to submit PRs for the docs if this is helpful, but I did notice some documentation PRs from January that are languishing. I'm just happy to help the project in any way I can.

**Comment by davehorton** [comment]
Yes a PR for the docs would be great: https://github.com/davehorton/drachtio-site/tree/master/docs/drachtio-server <https://github.com/davehorton/drachtio-site/tree/master/docs/drachtio-server>

On Jun 4, 2020, at 4:57 AM, Moonlit Knight <notifications@github.com> wrote:


Hi Dave, as with calvet27, my problem was solved by replacing sip with sips.
My comment was that the documentation says sip so is this a documentation error?

I'm just starting with drachtio and would be happy to submit PRs for the docs if this is helpful, but I did notice some documentation PRs from January that are languishing. I'm just happy to help the project in any way I can.
